backend/contract-service/src/main/java/org/egov/works/service/NotificationService.java

```diff
@@ -0,0 +1,336 @@
+        String shortendURL = getShortnerURL(CBOUrl.toString());
+
+        DateFormat formatter = new SimpleDateFormat("dd/MM/yyyy");
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTimeInMillis(dueDate);
+        String date = formatter.format(calendar.getTime());
+
+        message = message.replace("{contactPersonName}", userDetailsForSMS.get("personName"))
+                .replace("{organisationName}", userDetailsForSMS.get("orgName"))
+                .replace("{IA/IP}", contract.getExecutingAuthority())
+                .replace("{PROJECT_NAME}", userDetailsForSMS.get("projectName"))
+                .replace("{CONTRACT_NUMBER}", contract.getContractNumber())
+                .replace("{dueDate}", formatter.format(calendar.getTime()))
+                .replace("{Organization_Login_URL}", shortendURL);
+        return message;
+    }
+
+    public String buildMessageForApproveAction_WOCreator(Contract contract, Map<String, String> userDetailsForSMS, String message) {
+        message = message.replace("{CONTRACT_NUMBER}", contract.getContractNumber())
+                .replace("{PROJECT_NAME}", userDetailsForSMS.get("projectName"))
+                .replace("{LOCATION}", userDetailsForSMS.get("locationName"))
+                .replace("{organisationName}", userDetailsForSMS.get("orgName"));
+        return message;
+    }
+
+    /**
+     * Creates a cache for localization that gets refreshed at every call.
+     *
+     * @param requestInfo
+     * @param rootTenantId
+     * @param locale
+     * @param module
+     * @return
+     */
+    public Map<String, Map<String, String>> getLocalisedMessages(RequestInfo requestInfo, String rootTenantId, String locale, String module) {
+        Map<String, Map<String, String>> localizedMessageMap = new HashMap<>();
+        Map<String, String> mapOfCodesAndMessages = new HashMap<>();
+        StringBuilder uri = new StringBuilder();
+        RequestInfoWrapper requestInfoWrapper = new RequestInfoWrapper();
+        requestInfoWrapper.setRequestInfo(requestInfo);
+        uri.append(config.getLocalizationHost()).append(config.getLocalizationContextPath())
+                .append(config.getLocalizationSearchEndpoint()).append("?tenantId=" + rootTenantId)
+                .append("&module=" + module).append("&locale=" + locale);
+        List<String> codes = null;
+        List<String> messages = null;
+        Object result = null;
+        try {
+            result = repository.fetchResult(uri, requestInfoWrapper);
+            codes = JsonPath.read(result, ContractServiceConstants.CONTRACTS_LOCALIZATION_CODES_JSONPATH);
+            messages = JsonPath.read(result, ContractServiceConstants.CONTRACTS_LOCALIZATION_MSGS_JSONPATH);
+        } catch (Exception e) {
+            log.error("Exception while fetching from localization: " + e);
+        }
+        if (null != result) {
```

Comment:
we should use result != null

Reply:
null!=result is more accurate as it prevents null pointer Exception. I read about this and then used this change